runtimebp/cpu.go

```diff
@@ -55,6 +56,40 @@ func NumCPU() (n float64) {
 	return quota / period
 }
 
+// On some really old docker version the quota file will be -1, in which case we
+// should use this one instead.
+func numCPUSharesFallback() (n float64) {
+	const (
+		sharesPath  = "/sys/fs/cgroup/cpu/cpu.shares"
+		denominator = 1024
+	)
+
+	var err error
+	defer func() {
+		if err != nil || n <= 0 {
+			// Fallback and log to stderr.
+			fmt.Fprintf(
+				os.Stderr,
+				"NumCPU: falling back to use runtime.NumCPU(): %v, %v\n",
+				n,
+				err,
+			)
+			n = float64(runtime.NumCPU())
+		}
+	}()
+
+	// Big enough buffer to read the number in the file wholly into memory.
+	buf := make([]byte, 1024)
```

Comment:
Why not just do a `ioutil.ReadFile` and convert that?

Reply:
the file could be huge, we don't want to blow up memory.

Reply:
Doesn't that file only ever have a number in there? Well, doesn't matter, was just curious :)

Reply:
It could be misconfigured somewhere.